cukinia

```diff
@@ -363,16 +363,45 @@ _cukinia_process()
 {
 	local pname="$1"
 	local puser="$2"
-	local ret
+	local proc_name
+	local proc_uid
 
 	_cukinia_prepare "Checking process \"$pname\" as ${puser:-any user}"
-	ret=$(ps -a -o user,comm |
-			awk '($2 == "'$pname'") && ($1 ~ /^'${puser:-.*}'$/) {
-			    print;
-			}')
-	[ -n "$ret" ] && return 0 || return 1
-}
 
+	if [ ! -z $puser ]; then
+		puser_id=`id -u $puser &>/dev/null`
+		# check if owner exist, return 1 then
+		if [ $? -ne 0 ]; then
```

Comment:
Honestly, I really dislike such tests in shell (`if [ $? -ne 0 ]`).

I do prefer testing the command inside the if statement. `if ! puser_uid="$(id -u "$puser" 2>/dev/null)"; then`